addons/mail/models/mail_activity_type.py

```diff
@@ -125,3 +125,51 @@ def _inverse_triggered_next_type_id(self):
                 activity_type.chaining_type = 'trigger'
             else:
                 activity_type.chaining_type = 'suggest'
+
+    def write(self, values):
+        """ Protect some master types against deletiong or modification when they
+        are used as default in apps, in critical flows, plans, ... """
+        if values.get('res_model'):
+            xmlid_to_model = self._get_xmlid_model_link()
+            modified = self.browse()
+            for xml_id, model in xmlid_to_model.items():
+                activity_type = self.env.ref(xml_id, raise_if_not_found=False)
+                if activity_type and values['res_model'] != model:
+                    modified |= activity_type
+            modified = modified & self
+            if modified:
+                raise exceptions.UserError(
+                    _('You cannot modify %(activities_names)s target model as those are required in various apps.',
+                      activities_names=', '.join(act.name for act in modified),
+                ))
+        return super().write(values)
+
+    @api.model
+    def _get_xmlid_model_link(self):
+        """ Force a model for some xml ids of activity types. """
+        return {
+            'mail.mail_activity_data_call': False,  # generic call, used notably in VOIP, ...
+            'mail.mail_activity_data_meeting': False,  # generic meeting, used in calendar, hr, ...
+            'mail.mail_activity_data_todo': False,  # generic todo, used in plans, ...
+            'mail.mail_activity_data_upload_document': False,  # generic upload, used in documents, accounting, ...
```

Comment:
Small note: this one does not seem to have much usage in current master, maybe it was replaced by something else? Would be interesting to check during FW.